### PR TITLE
Project2 phase2

### PR DESCRIPTION
**Pull request by Mazen-Barakat**

### CODE REVIEW

**Comment by amrbahaa1** [comment]
Course details page is not working -10
search bar is not working -10
no header and footer - 10

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
 import "./App.css";
-import CardsSection from './components/CoursesSection/CardsSection';
+import Navbar from "./components/Navbar";
+import Home from "./pages/Home";
+import CoursePage from "./pages/CoursePage";
+import Header from "./components/Header";
+import CoursesSection from "./components/CoursesSection";
```
Comment (amrbahaa1) [inline src/App.js:6]: Line 5:8:   'Header' is defined but never used          no-unused-vars
  Line 6:8:   'CoursesSection' is defined but never used  no-unused-vars
  -3

File: src/App.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
 import "./App.css";
-import CardsSection from './components/CoursesSection/CardsSection';
+import Navbar from "./components/Navbar";
```
Comment (amrbahaa1) [inline src/App.js:2]: 'Navbar' is defined but never used

File: src/App.js [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+import Home from "./pages/Home";
+import CoursePage from "./pages/CoursePage";
+import Header from "./components/Header";
+import CoursesSection from "./components/CoursesSection";
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
```
Comment (amrbahaa1) [inline src/App.js:7]: 'Navigate' is defined but never used

File: src/components/CourseContent.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import "./styles/CourseContent.css";
+
+function CourseContent(props) {
+  const [showMore, setShowMore] = useState(false);
+  const numberOfSections = showMore
+    ? props?.courseData?.curriculum_context?.data?.sections?.length
+    : Math.min(
+        10,
+        props?.courseData?.curriculum_context?.data?.sections?.length
+      );
```
Comment (amrbahaa1) [inline src/components/CourseContent.js:11]: not readable 
-3

File: src/components/CourseHeader.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+function CourseHeader(props) {
+  return (
+    <div className="courseheader">
+      <div className="headerContent">
+        <p className="path">
+          <a>Development</a>
+          {"   >   "}
+          <a> Programming Languages</a>
+          {"   >   "}
+          <a>{props?.courseSummary?.context_info?.label?.display_name}</a>
```
Comment (amrbahaa1) [inline src/components/CourseHeader.js:13]: The href attribute is required for an anchor to be keyboard accessible. Provide a valid, navigable address as the href value. If you cannot provide an href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md
 -3

File: src/components/CourseHeader.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        </p>
+        <img className="imge" src={props?.courseSummary?.image_125_H} />
```
Comment (amrbahaa1) [inline src/components/CourseHeader.js:15]: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images
-1

**Review by amrbahaa1 — CHANGES_REQUESTED** [review]
the code is not clean code enough and some functionalities are not working 
overall grade 60 / 100

**Comment by Mazen-Barakat** [comment]
> Course details page is not working -10 search bar is not working -10 no header and footer - 10

**Comment by Mazen-Barakat** [comment]
> Course details page is not working -10 search bar is not working -10 no header and footer - 10

The footer is on the course page. What is the problem?

**Pull request closed without merging**